File: iosApp/iosApp/ui/payment/PaymentMethodUiState.swift
import Shared

struct PaymentMethodUiState {
    var isLoading: Bool = true

    var selected: PaymentMethod?

    // 제출 in-flight — 중복 호출 방지 (규칙 §9)
    var isSubmitting: Bool = false

    var canSubmit: Bool {
        !isSubmitting && selected != nil
    }
}

File: composeApp/src/commonMain/kotlin/org/sesacteamproject/passmate/ui/payment/PaymentMethodUiState.kt
package org.sesacteamproject.passmate.ui.payment

import org.sesacteamproject.passmate.payment.domain.model.PaymentMethod

data class PaymentMethodUiState(
    val isLoading: Boolean = true,
    val selected: PaymentMethod? = null,
    // 제출 in-flight — 중복 호출 방지 (규칙 §9)
    val isSubmitting: Boolean = false
) {

    val canSubmit: Boolean
        get() = !isSubmitting && selected != null
}
